source/browseMode.py

```diff
@@ -628,6 +628,25 @@ def addQuickNav(
 		if key is not None:
 			cls.__gestures["kb:shift+%s" % key] = scriptName
 
+	@classmethod
+	def _addQuickNavHeading(
+		cls,
+		levelRange: range,
+	):
+		for i in levelRange:
```

Comment:
Could we add some constraints here? For instance, non-positive heading levels don't make sense, and I don't see how we can support heading levels over 9.
```suggestion
		for i in levelRange:
			if not (0 < i < 10):
				log.debug(f"Skipping adding quick navigation key for heading level {i}.")
				continue
```